app/registries/foreman/plugin.rb

```diff
@@ -372,6 +372,11 @@ def register_facet(klass, name, &block)
       @facets << Facets.register(klass, name, &block)
     end
 
+    def register_info_provider(klass)
+      @info_providers ||= []
+      @info_providers << HostInfo.register_info_provider(klass)
```

Comment:
why calling `HostInfo.register_info_provider(klass)`? I thought this plugin instance is keeping the list and HostInfo asks it  here https://github.com/theforeman/foreman/pull/3701/files#diff-adbf4db3b5ffb094ecec3a7a4f1f841bR12